pkg/broker/handler/handler_test.go

```diff
@@ -92,39 +92,50 @@ func TestHandler(t *testing.T) {
 	testEvent.SetSubject("subject")
 	testEvent.SetType("type")
 
-	if err := p.Send(ctx, binding.ToMessage(&testEvent)); err != nil {
-		t.Errorf("failed to seed event to pubsub: %v", err)
-	}
-	gotEvent := nextEventWithTimeout(eventCh)
-	if diff := cmp.Diff(&testEvent, gotEvent); diff != "" {
-		t.Errorf("processed event (-want,+got): %v", diff)
-	}
-
-	// TODO(yolocs): when nack is ready.
-	// processor.oneTimeErr = true
-	// go func() {
-	// 	assertNextEvent(t, &testEvent, eventCh)
-	// 	assertNextEvent(t, &testEvent, eventCh)
-	// }()
-	// if err := p.Send(ctx, binding.ToMessage(&testEvent)); err != nil {
-	// 	t.Errorf("failed to seed event to pubsub: %v", err)
-	// }
-
-	unlock := processor.Lock()
-	processor.BlockUntilCancel = true
-	unlock()
-	if err := p.Send(ctx, binding.ToMessage(&testEvent)); err != nil {
-		t.Errorf("failed to seed event to pubsub: %v", err)
-	}
-	gotEvent = nextEventWithTimeout(eventCh)
-	if diff := cmp.Diff(&testEvent, gotEvent); diff != "" {
-		t.Errorf("processed event (-want,+got): %v", diff)
-	}
-	unlock = processor.Lock()
-	if !processor.WasCancelled {
-		t.Error("processor was not cancelled on timeout")
-	}
-	unlock()
+	t.Run("handle event success", func(t *testing.T) {
+		if err := p.Send(ctx, binding.ToMessage(&testEvent)); err != nil {
+			t.Errorf("failed to seed event to pubsub: %v", err)
+		}
+		gotEvent := nextEventWithTimeout(eventCh)
+		if diff := cmp.Diff(&testEvent, gotEvent); diff != "" {
+			t.Errorf("processed event (-want,+got): %v", diff)
+		}
+	})
+
+	t.Run("retry event on processing failure", func(t *testing.T) {
+		unlock := processor.Lock()
+		processor.OneTimeErr = true
```

Comment:
I didn't use table test because I think the way each verifies result is different - retry needs to verify the event was received twice and timeout needs to verify the processing was actually cancelled. Use `t.Run` is just for informational purpose. If you'd prefer, I can make them separate test cases.

Reply:
That would be my preference, but it isn't a strong one.